### PR TITLE
[PIR][DynamicShape] Add InferSymbolicShape for arange, expand

### PR DESCRIPTION
**Pull request by lanxianghit**
### PR types

Others 

### PR changes

Others 

### Description

Pcard-67164
Add InferSymbolicShape for arange, expand

### CODE REVIEW

File: paddle/fluid/pir/dialect/operator/interface/infer_symbolic_shape.cc [hunk 1]
```diff
@@ -1001,8 +1001,54 @@ bool Relu_OpInferSymbolicShape(pir::Operation *op,
 
 bool ArangeOpInferSymbolicShape(
     pir::Operation *op, pir::ShapeConstraintIRAnalysis *shape_analysis) {
-  PADDLE_THROW(phi::errors::Unimplemented(
-      op->name() + " DOES NOT have InferSymbolicShapeInterface!"));
+  const auto start_shape_or_data =
```
Comment (tc20042008) [inline paddle/fluid/pir/dialect/operator/interface/infer_symbolic_shape.cc]: const auto&

Comment (lanxianghit) [reply paddle/fluid/pir/dialect/operator/interface/infer_symbolic_shape.cc]: done

File: paddle/fluid/pir/dialect/operator/interface/infer_symbolic_shape.cc [hunk 2]
```diff
@@ -1001,8 +1001,54 @@ bool Relu_OpInferSymbolicShape(pir::Operation *op,
...
 bool ArangeOpInferSymbolicShape(
     pir::Operation *op, pir::ShapeConstraintIRAnalysis *shape_analysis) {
-  PADDLE_THROW(phi::errors::Unimplemented(
-      op->name() + " DOES NOT have InferSymbolicShapeInterface!"));
+  const auto &start_shape_or_data =
+      shape_analysis->GetShapeOrDataForValue(op->operand_source(0));
+  const auto &end_shape_or_data =
+      shape_analysis->GetShapeOrDataForValue(op->operand_source(1));
+  const auto &step_shape_or_data =
+      shape_analysis->GetShapeOrDataForValue(op->operand_source(2));
```
Comment (tc20042008) [inline paddle/fluid/pir/dialect/operator/interface/infer_symbolic_shape.cc:1009]: 这些变量可以继续放到各自的lambda init里去

Comment (tc20042008) [reply paddle/fluid/pir/dialect/operator/interface/infer_symbolic_shape.cc:1009]: 下同

**Review by tc20042008 — APPROVED** [review]

File: paddle/fluid/pir/dialect/operator/interface/infer_symbolic_shape.cc (hunk 2 above)
Comment (lanxianghit) [reply paddle/fluid/pir/dialect/operator/interface/infer_symbolic_shape.cc:1009]: thx，可以后续的PR统一改一下，免得ci重新排队了

**Pull request merged**